Sources/SQL/Query/DeclaredField.swift

```diff
@@ -0,0 +1,200 @@
+    return DeclaredField(name: name)
+}
+
+public extension Sequence where Iterator.Element == DeclaredField {
+    public func queryComponentsForSelectingFields(useQualifiedNames useQualified: Bool, useAliasing aliasing: Bool, isolateQueryComponents isolate: Bool) -> QueryComponents {
+        let string = map {
+            field in
+
+            var str = useQualified ? field.qualifiedName : field.unqualifiedName
+
+            if aliasing && field.qualifiedName != field.alias {
+                str += " AS \(field.alias)"
+            }
+
+            return str
+        }.joined(separator: ", ")
+
+        if isolate {
+            return QueryComponents(string).isolate()
+        }
+        else {
+            return QueryComponents(string)
+        }
+    }
+}
+
+public extension Collection where Iterator.Element == (DeclaredField, Optional<SQLData>) {
+    public func queryComponentsForSettingValues(useQualifiedNames useQualified: Bool) -> QueryComponents {
+        let string = map {
+            (field, value) in
+
+            var str = useQualified ? field.qualifiedName : field.unqualifiedName
+
+            str += " = " + QueryComponents.valuePlaceholder
+
+            return str
+
+        }.joined(separator: ", ")
+
+        return QueryComponents(string, values: map { $0.1 })
+    }
+
+    public func queryComponentsForValuePlaceHolders(isolated isolate: Bool) -> QueryComponents {
+        let string = map {
+            (_, value) in
+                guard let value = value else {
+                    return QueryComponents.valuePlaceholder
+                }
+                switch value {
+                case let .RawSQL(rawSql):
+                    return rawSql
+                default:
+                    return QueryComponents.valuePlaceholder
+                }
+        }.joined(separator: ", ")
+
+        let components = QueryComponents(string, values: map { $0.1 })
+
+        return isolate ? components.isolate() : components
+    }
+}
+
+public func == (lhs: DeclaredField, rhs: DeclaredField) -> Bool {
+    return lhs.hashValue == rhs.hashValue
+}
+
+public extension DeclaredField {
+
+    public var qualifiedName: String {
+        guard let tableName = tableName else {
+            return unqualifiedName
+        }
+
+        return tableName + "." + unqualifiedName
+    }
+
+    public var alias: String {
+        if let aliasName = aliasName {
+            return aliasName
+        }
+//        guard let tableName = tableName else { //todo fix it
+        return unqualifiedName
+//        }
+
+//        return tableName + "__" + unqualifiedName
+    }
+
+    public func alias(newAliasName: String) -> DeclaredField {
+        var new = self
+        new.aliasName = newAliasName
+        return new
+    }
+
+    public var description: String {
+        return qualifiedName
+    }
+
+    public func containedIn<T: SQLDataConvertible>(values: [T?]) -> Condition {
```

Comment:
in first commit I divided model  from table and moved code to other file, git show it as added lines. I just moved it. Other words - it's old code. Will change in future